src/vaibhav/Shop.java

```diff
@@ -0,0 +1,66 @@
+package vaibhav;
+
+public class Shop {	
+	int maggie = 50;
+	int dosa = 43;
+	int oilPackets = 39;
+	int paniPuri = 43;
+	int masala = 73;
+	
+	void shopping(int noOfMaggies,int noOfDosas,int noOfOilPackets,int NoOfPaniPuri,int noOfMasalas ) {		
+		if(noOfMaggies>maggie)
+			maggie = 0;
```

Comment:
if we passing noOfMaggies as 51 then you are making Maggie = 0, which looks logically incorrect